lib/bemhtml/runtime/match.js

```diff
@@ -216,6 +224,9 @@ Match.prototype.exec = function exec(context) {
   this.bemhtml.match = oldMatch;
   this.restoreDepth(save);
 
+  if (out instanceof Error)
```

Comment:
This should check if `.thrownError` is not `null`, load it, cleanup the field (set to `null`), and throw  it.